sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/AnsiTypeCoercion.scala

```diff
@@ -178,6 +178,10 @@ object AnsiTypeCoercion extends TypeCoercionBase {
       case (StringType, DecimalType) if isInputFoldable =>
         Some(DecimalType.SYSTEM_DEFAULT)
 
+      // If the target type is Decimal type and it wider than input, cast the input to decimal.
```

Comment:
On second thought. This code should not be needed.
It will be covered by line 189
```
case (n1: NumericType, n2: NumericType) =>
        val widerType = findWiderTypeForTwo(n1, n2)
         ....
```

We don't need to touch `AnsiTypeCoercion.scala`